installer/x86_64/install.sh

```diff
@@ -424,6 +451,12 @@ fi
 # Decompress the file for the file system directly to the partition
 unzip -o $ONIE_INSTALLER_PAYLOAD -x "$FILESYSTEM_DOCKERFS" -d $demo_mnt/$image_dir
 
+# Indicate that this filesystem is being generated using the installer on the build server
+# to facilitate migration from a 3rd party OS into SONiC
+if [ "$install_env" = "build" ]; then
+   touch $demo_mnt/$image_dir/platform/migration
+fi
```

Comment:
I prefer not to introduce this migration flag. In the rc.local, if it is the first time, check if there is /host/machine.conf or not first, if there is no /host/machine.conf, then mount onie and copy the file. We do not really need the migration flag here.